lib/src/ruby/ruby_value.cc

```diff
@@ -87,7 +87,10 @@ namespace facter { namespace ruby {
             return;
         }
         if (ruby.is_fixednum(value)) {
-            json.SetInt64(ruby.rb_num2long(value));
+            // On Windows 64-bit with Ruby 2.2, rb_num2long returns the wrong value for negative numbers.
+            // FixedNums can only represent a word (32-bit) so explicitly cast to an int.
+            // TODO: Handle BigNum.
+            json.SetInt64(static_cast<int>(ruby.rb_num2long(value)));
```

Comment:
Looking at the impl, this function can handle big nums too.
On Apr 1, 2016 10:45 AM, "Michael Smith" <notifications@github.com> wrote:

> In lib/src/ruby/ruby_value.cc
> <https://github.com/puppetlabs/facter/pull/1295#discussion_r58241093>:
>
> > @@ -87,7 +87,10 @@ namespace facter { namespace ruby {
> >              return;
> >          }
> >          if (ruby.is_fixednum(value)) {
> > -            json.SetInt64(ruby.rb_num2long(value));
> > +            // On Windows 64-bit with Ruby 2.2, rb_num2long returns the wrong value for negative numbers.
> > +            // FixedNums can only represent a word (32-bit) so explicitly cast to an int.
> > +            // TODO: Handle BigNum.
> > +            json.SetInt64(static_cast<int>(ruby.rb_num2long(value)));
>
> I'll give it a try. The behavior in general seems weird.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/facter/pull/1295/files/77c282bbec4b58e315304a0a1cd2ca9e01900e88#r58241093>
>